packaging/prerelease/s3_index.sh

```diff
@@ -5,35 +5,35 @@ set -e -u -o pipefail # Fail on error
 dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 cd $dir
 
+# If editing this file, you may also want to update windows/s3_prerelease.cmd
+
 client_dir="$dir/../.."
 bucket_name=${BUCKET_NAME:-}
-save_dir="/tmp/s3index"
+platform=${PLATFORM:-}
 
 if [ "$bucket_name" = "" ]; then
   echo "No BUCKET_NAME"
   exit 1
 fi
 
+if [ "$platform" = "" ]; then
+  echo "No PLATFORM"
+  exit 1
+fi
+
 echo "Loading release tool"
 "$client_dir/packaging/goinstall.sh" "github.com/keybase/release"
 release_bin="$GOPATH/bin/release"
 
-# Clear files
-rm -rf "$save_dir"
-mkdir -p "$save_dir"
-
 echo "Creating index files"
-"$release_bin" index-html --bucket-name="$bucket_name" --prefixes="darwin/,linux_binaries/deb/,linux_binaries/rpm/,windows/" --dest="$save_dir/index.html"
-"$release_bin" index-html --bucket-name="$bucket_name" --prefixes="electron-sourcemaps/" --dest="$save_dir/electron-sourcemaps/index.html"
-
-echo "Syncing index files"
-s3cmd sync --acl-public --disable-multipart $save_dir/* s3://$bucket_name/
+"$release_bin" index-html --bucket-name="$bucket_name" --prefixes="darwin/,linux_binaries/deb/,linux_binaries/rpm/,windows/" --upload="index.html"
+"$release_bin" index-html --bucket-name="$bucket_name" --prefixes="electron-sourcemaps/" --upload="electron-sourcemaps/index.html"
 
-echo "Linking latest"
-"$release_bin" latest --bucket-name="$bucket_name"
+echo "Linking latest ($platform)"
+"$release_bin" latest --bucket-name="$bucket_name" --platform="$platform"
 
-echo "Checking test releases"
-"$release_bin" promote-test-releases --bucket-name="$bucket_name"
+echo "Checking if we need to promote a release for testing ($platform)"
+"$release_bin" promote-test-releases --bucket-name="$bucket_name" --platform="$platform"
 
-echo "Checking releases"
-"$release_bin" promote-releases --bucket-name="$bucket_name"
+echo "Checking if we need to promote a release ($platform)"
+"$release_bin" promote-releases --bucket-name="$bucket_name" --platform="$platform"
```

Comment:
For linux, promote-test-releases copies update-linux-prod.json to update-linux-prod-test.json .... so admin users and public users get the same build...

The promote-releases is a no-op (since there is nothing to promote):
https://github.com/keybase/release/blob/master/update/s3.go#L497